source/MaterialXRuntime/RtFileIo.cpp

```diff
@@ -1195,6 +1196,56 @@ namespace
         }
     }
 
+    void writeMasterPrim(DocumentPtr document, RtStagePtr rtStage, const RtPrim& masterPrim)
+    {
+        if (!masterPrim.isValid())
+        {
+            return;
+        }
+
+        // Write the definition
+        PvtStage* stage = PvtStage::ptr(rtStage);
```

Comment:
I see that you are getting the private stage and prim pointers, are you sure this is a good way to do this? I would think these should only be accessed in the RtStage and RtPrim classes, no? Maybe additional functions need to be exposed on these classes to do what you want?

Reply:
On second thought we seem to do this throughout RtFileIo.cpp, maybe we should chat offline about this.